### PR TITLE
 [SR SQL Planner Summer Camp] Support ShowRoutineLoadStmt with new parser and analyzer

### LINKED ISSUE #7949: New Parser and Analyzer support RoutineLoad-related statements （Challenging）
# Required work and output

1.  Use antlr4 to complete parse statements. The generated statement must follow the original data structure, and should be compatible with the old statement.
    
2.  In the original syntax file  **sql_parser.cup**, the new syntax needs to be compatible with the syntax of the old version, using the syntax file sql_parser.cup of the old version as the reference standard.
    
3.  Remove the logic of analytic functions in the statement and move the relevant logic to the new Analyzer framework. The new Analyzer is in the SQL/Analyzer directory.
    
4.  If the code related to Privilege Check exists in the original analyzer, move the code related to privilege control into PrivilegeChecker.
    
5.  When submitting the PR, please make sure there is unit test coverage for related changes.
    

# Tasks

 - [ ] AlterRoutineLoadStmt
 - [ ] PauseRoutineLoadStmt
 - [ ] StopRoutineLoadStmt
 - [ ] CreateRoutineLoadStmt
 - [ ] ResumeRoutineLoadStmt
 - [ ] ShowRoutineLoadStmt
 - [ ] ShowRoutineLoadTaskStmt

# How to participate?

1.  Pick an issue you are interested in, click the issue link and leave a comment under the issue to sign up.  **One person can only choose one issue and the first person who picks the issue will be assigned.**
    
2.  **After the issue is assigned, you should complete the task in 14 days.** If you are unable to complete it within the specified time, the task will be withdrawn and assigned to others.
    
3.  Participants are required to produce two pieces of content (forms including but not limited to mind maps, study notes, blogs, and tutorials) in the course of the activity, each of which should be at least 200 words.
    
4.  Got questions? Join our  [Slack channel](https://join.slack.com/t/starrocks/shared_invite/zt-192zeqlc7-9MAvqAXNcveW_l5Pqr8r2g)  and find channel:  **#sql-planner**.
    

# Before you start

1.  Read  [Contributing.md](https://github.com/StarRocks/starrocks/blob/main/CONTRIBUTING.md)  carefully to understand how to submit a pull request and how to configure BE and FE development environment.
    
2.  Sign the  [Contributor License Agreement](https://cla-assistant.io/StarRocks/starrocks)  before you make a pull request.
    
3.  To learn more about StarRocks, please refer to  [StarRocks documentation](https://github.com/StarRocks/docs/blob/main/TOC.md).
    

# Pull request requirements

1.  Create a pull request (PR) after you complete the task.
* PR title format: [SR SQL Planner Summer Camp] + PR title
2. After submitting PR, you can comment under the pinned Issue [SR SQL Planner Summer Camp].
* Comment format: "I have finished Issue #" + PR link

### PR DESCRIPTION
**Pull request by shshenhua**
## What type of PR is this：
- [ ] feature

## Which issues of this PR fixes ：

Fixes  https://github.com/StarRocks/starrocks/issues/7949

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 19 / 60 (31.67%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|0|18|00.00%|[1154, 1155, 1156, 1157, 1159, 1160, 1161, 1163, 1164, 1165, 1167, 1168, 1169, 1170, 1172, 1173, 1174, 1176]|
|:large_blue_circle:|com/starrocks/analysis/FunctionalExprProvider.java|0|7|00.00%|[145, 147, 148, 149, 151, 153, 154]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|0|4|00.00%|[834, 835, 836, 838]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|0|1|00.00%|[936]|
|:large_blue_circle:|com/starrocks/analysis/ShowRoutineLoadStmt.java|14|25|56.00%|[133, 139, 140, 142, 177, 184, 187, 188, 189, 192, 228]|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/ShowStmtAnalyzer.java|4|4|100.00%|[]|

**Comment by imay** [comment]
@mergify rebase

File: fe/fe-core/src/test/java/com/starrocks/analysis/ShowRoutineLoadStmtTest.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
+// This file is made available under Elastic License 2.0.
```
Comment (imay) [inline fe/fe-core/src/test/java/com/starrocks/analysis/ShowRoutineLoadStmtTest.java:1]: new files should use starrocks license header
// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 35 / 91 (38.46%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|0|25|00.00%|[1155, 1156, 1157, 1159, 1160, 1161, 1163, 1168, 1169, 1170, 1171, 1173, 1174, 1175, 1177, 1178, 1179, 1181, 1182, 1183, 1184, 1186, 1187, 1188, 1190]|
|:large_blue_circle:|com/starrocks/analysis/FunctionalExprProvider.java|0|7|00.00%|[145, 147, 148, 149, 151, 153, 154]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|0|4|00.00%|[834, 835, 836, 838]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|0|1|00.00%|[936]|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|1|2|50.00%|[314]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PauseRoutineLoadAnalyzer.java|7|13|53.85%|[13, 18, 19, 20, 22, 23]|
|:large_blue_circle:|com/starrocks/analysis/ShowRoutineLoadStmt.java|14|25|56.00%|[133, 139, 140, 142, 177, 184, 187, 188, 189, 192, 228]|
|:large_blue_circle:|com/starrocks/analysis/PauseRoutineLoadStmt.java|7|8|87.50%|[58]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/ShowStmtAnalyzer.java|4|4|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 19 / 60 (31.67%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|0|18|00.00%|[1154, 1155, 1156, 1157, 1159, 1160, 1161, 1163, 1164, 1165, 1167, 1168, 1169, 1170, 1172, 1173, 1174, 1176]|
|:large_blue_circle:|com/starrocks/analysis/FunctionalExprProvider.java|0|7|00.00%|[145, 147, 148, 149, 151, 153, 154]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|0|4|00.00%|[834, 835, 836, 838]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|0|1|00.00%|[936]|
|:large_blue_circle:|com/starrocks/analysis/ShowRoutineLoadStmt.java|14|25|56.00%|[133, 139, 140, 142, 177, 184, 187, 188, 189, 192, 228]|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/ShowStmtAnalyzer.java|4|4|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/analysis/ShowRoutineLoadStmt.java [hunk 2]
```diff
@@ -119,20 +126,31 @@ public boolean isIncludeHistory() {
         return includeHistory;
     }
 
-    public RoutineLoadFunctionalExprProvider getFunctionalExprProvider() {
+    public RoutineLoadFunctionalExprProvider getFunctionalExprProvider(ConnectContext context) throws AnalysisException {
         if (null == functionalExprProvider) {
             functionalExprProvider = new RoutineLoadFunctionalExprProvider();
         }
+        functionalExprProvider.analyze(context, whereClause, orderElements, limitElement);
```
Comment (HangyuanLiu) [inline fe/fe-core/src/main/java/com/starrocks/analysis/ShowRoutineLoadStmt.java:133]: can't add analyze code here

**Pull request closed without merging**